src/sentry/web/frontend/base.py

```diff
@@ -128,111 +128,109 @@ def respond(
+        )
+    else:
+        return None
 
-    # TODO(dcramer): move the implicit organization logic into its own class
-    # as it's only used in a single location and over complicates the rest of
-    # the code
-    def determine_active_organization(
-        self, request: HttpRequest, organization_slug: str | None = None
-    ) -> None:
-        """
-        Using the current request and potentially optional organization_slug, 'determines'
-        the current session for this mixin object's scope, placing it into the active_organization attribute.
 
-        Generally this method only need be called once at the head of a request, as it can potentially have side
-        effects in the user's session.  That said, when login occurs during a request, this method should be called
-        be called again to refresh an active organization context.
-        """
+def _try_finding_org_from_slug(
+    is_implicit: bool,
+    organization_slug: str,
+    organizations: list[RpcOrganizationSummary],
+    request: HttpRequest,
+) -> RpcUserOrganizationContext | None:
+    try:
+        backup_org = next(o for o in organizations if o.slug == organization_slug)
+    except StopIteration:
+        logger.info("Active organization [%s] not found in scope", organization_slug)
+        if is_implicit:
+            session = request.session
+            if session and "activeorg" in session:
+                del session["activeorg"]
+        backup_org = None
+
+    if backup_org is not None:
+        return organization_service.get_organization_by_id(
+            id=backup_org.id, user_id=request.user.id
+        )
+    return None
+
+
+def _lookup_organizations(
+    is_implicit: bool, organization_slug: str | None, request: HttpRequest
+) -> tuple[RpcUserOrganizationContext | None, RpcOrganizationSummary | None]:
+    active_organization: RpcUserOrganizationContext | None = _try_superuser_org_lookup(
+        organization_slug, request
+    )
+    backup_organization: RpcOrganizationSummary | None = None
+    if active_organization is None and request.user.id is not None:
+        organizations = user_service.get_organizations(user_id=request.user.id, only_visible=True)
+
+        if organizations:
+            backup_organization = organizations[0]
+            if organization_slug:
+                active_organization = _try_finding_org_from_slug(
+                    is_implicit, organization_slug, organizations, request
+                )
+    return active_organization, backup_organization
 
-        if organization_slug is None:
-            is_implicit = True
-            organization_slug = self._find_implicit_slug(request)
-        else:
-            is_implicit = False
 
-        active_organization, backup_organization = self._lookup_organizations(
-            is_implicit, organization_slug, request
-        )
+# TODO(dcramer): move the implicit organization logic into its own class
```

Comment:
can remove this todo, no?

Reply:
it still happens in the callers of this function annoyingly so I'm leaving it for now -- even though personally I don't think the TODO is a good idea 😆